modules/common/secret/secret.go

```diff
@@ -385,3 +386,70 @@ func GetDataFromSecret(
 
 	return data, ctrl.Result{}, nil
 }
+
+// EnsureSecretData - ensures that the Secret object exists and the expected fields
+// are in the Secret. It returns a hash of the values of the expected fields.
+func EnsureSecretData(
+	ctx context.Context,
+	secretName types.NamespacedName,
+	expectedFields []string,
+	reader client.Reader,
```

Comment:
could just pass in a helper `h *helper.Helper`  which has the client.

Reply:
it also has a logger

Reply:
> could just pass in a helper `h *helper.Helper` which has the client.

I would rather pass the interface than the full Helper. The Helper is bigger surface than what we need here.

Reply:
> it also has a logger

I would rater use logger.FromContext(ctx) to get a logger. As we discussed before that logger has everything we need.

Reply:
right, I did not mean to use the logging funcs from lib-common. Just that the helper already has an initialized logger from the reconcile instance, which could be `logger.FromContext(ctx)` . Also just wanted to mention that the helper has the client and the logger which gets passed via dedicated params, so in general its possible to pass the helper instead of list of dedicated params.

Reply:
In general I don't like the Helper pattern as it couples unrelated things. Passing around the Helper means the function gets coupled to things that it does not need. I think the general good pattern would be to define these function parameters as interfaces. That will allow passing any object that matches the interface. So if it is called from a place that has just a client than it can pass that, but if it is called from a place that has a Helper it can still call the function by passing the client from the Helper to it.

Reply:
i agree with gibi here.

i would prefer to have an interface not use the helper stuct.
that applies to all our apis i don't think the helper stuct should really be in them in general even if that is what we pass.

on of the advantages of the interface apptoch si it will acutly be possible to implement a mock object that conform to the  interface and unit test the code.

that is much harder to do with concrete types.